src/HDF5.jl

```diff
@@ -222,6 +222,16 @@ const H5T_NATIVE_DOUBLE   = read_const(:H5T_NATIVE_DOUBLE_g)
 # Library versions
 const H5F_LIBVER_EARLIEST = 0
 const H5F_LIBVER_LATEST   = 1
+# Constructed types (occurs at runtime)
+function make_float16()
+    FLOAT16 = h5t_copy(H5T_NATIVE_FLOAT)
+    h5t_set_fields(FLOAT16, 15, 10, 5, 0, 10)
+    h5t_set_size(FLOAT16, 2)
+    h5t_set_ebias(FLOAT16, 15)
+    h5t_lock(FLOAT16)
+    return FLOAT16
+end
+# const H5T_FLOAT16 = make_float16()  (in `__init__()`)
```

Comment:
I have no idea if we can trigger a bug/make a test for this... would it happen if the HDF5 binary library was updated without this package being precompiled again? And yes, to me, filling references at run-time during `__init__` seems the safest possible thing to do.